istioctl/cmd/describe.go

```diff
@@ -1186,3 +1197,94 @@ func containerReady(pod *v1.Pod, containerName string) (bool, error) {
 	}
 	return false, fmt.Errorf("no container %q in pod", containerName)
 }
+
+// describePeerAuthentication fetches all PeerAuthentication in workload and root namespace.
+// It lists the ones applied to the pod, and the current active mTLS mode.
+// When the client doesn't have access to root namespace, it will only show workload namespace Peerauthentications.
+func describePeerAuthentication(writer io.Writer, configClient istioclient.Interface, istioNamespace, workloadNamespace string, pod *v1.Pod, podsLabels k8s_labels.Set) error { // nolint: lll
+	workloadPAList, err := configClient.SecurityV1beta1().PeerAuthentications(workloadNamespace).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to fetch workload namespace PeerAuthentication: %v", err)
+	}
+
+	hasRootNamespaceAccess := true
+	rootPAList, err := configClient.SecurityV1beta1().PeerAuthentications(istioNamespace).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		hasRootNamespaceAccess = false
+		fmt.Fprintf(writer, "Cannot analyze active mTLS mode since failed to fetch root namespace PeerAuthentication: %v:\n", err)
+	}
+
+	allPAs := append(rootPAList.Items, workloadPAList.Items...)
+	if len(allPAs) == 0 {
+		return nil
```

Comment:
I think it would be better to print the default mode (PERMISSIVE as returned by `ComposePeerAuthentication()`) even if there are no policies applied explicitlly.

Reply:
That makes sense, will do.